app/src/androidTest/java/com/epfl/drawyourpath/community/TournamentPostCreationViewTest.kt

```diff
@@ -0,0 +1,120 @@
+            TournamentPostCreationView()
+        }
+        // check default is first tournament in list
+        onView(withId(R.id.post_creation_tournament_spinner)).check(matches(withSpinnerText(TournamentPostCreationView.TOURNAMENT_SAMPLE[0])))
+        // click on spinner
+        onView(withId(R.id.post_creation_tournament_spinner)).perform(click())
+        // select third option
+        onData(anything()).atPosition(2).perform(click())
+        // check that third tournament is selected
+        onView(withId(R.id.post_creation_tournament_spinner)).check(matches(withSpinnerText(TournamentPostCreationView.TOURNAMENT_SAMPLE[2])))
+
+        scenario.close()
+    }
+
+    @Test
+    fun selectRunDisplayCorrectRun() {
+        val scenario = launchFragmentInContainer {
+            TournamentPostCreationView()
+        }
+
+        // wait for the runs to load
+        waitUntilAllThreadAreDone()
+        // check default is first run in list
+        onView(withId(R.id.post_creation_run_spinner)).check(matches(withSpinnerText(runs[0].toString())))
+        // click on spinner
+        onView(withId(R.id.post_creation_run_spinner)).perform(click())
+        // select second option
+        onData(anything()).atPosition(1).perform(click())
+        // check that second run is selected
+        onView(withId(R.id.post_creation_run_spinner)).check(matches(withSpinnerText(runs[1].toString())))
+
+        scenario.close()
+    }
+
+    @Test
+    fun backButtonGoesBackToCommunityView() {
+        val scenario = ActivityScenario.launch<MainActivity>(Intent(ApplicationProvider.getApplicationContext(), MainActivity::class.java))
+        // navigate to post creation
+        goToPostCreation()
+        // click on back button
+        onView(withId(R.id.post_creation_back_button)).perform(click())
+        // check that community fragment is displayed
+        onView(withId(R.id.community_add_post_button)).check(matches(isDisplayed()))
+
+        scenario.close()
+    }
+
+    @Test
+    fun postButtonGoesBackToTournamentView() {
+        val scenario = ActivityScenario.launch<MainActivity>(Intent(ApplicationProvider.getApplicationContext(), MainActivity::class.java))
+        // navigate to post creation
+        goToPostCreation()
+        // click on post button
+        onView(withId(R.id.post_creation_post_button)).perform(click())
+        // check that community fragment is displayed
+        onView(withId(R.id.community_add_post_button)).check(matches(isDisplayed()))
+
+        scenario.close()
+    }
+
+    private fun goToPostCreation() {
+        // go to community fragment
+        onView(withId(R.id.community_menu_item)).perform(click())
+        // go to post creation
+        onView(withId(R.id.community_add_post_button)).perform(click())
+    }
+
+    @Before
+    fun clearDatabase() {
+        InstrumentationRegistry.getInstrumentation().uiAutomation.executeShellCommand("pm clear com.epfl.drawyourpath.package").close()

```

Comment:
Maybe put the this line on the the top of the file since it will be executed first.